File: Golang/gin-songs-api/models/api/album.go
package api

import "time"

type Album struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Genre       string    `json:"genre"`
	Artist      Artist    `json:"artist"`
	ReleaseDate time.Time `json:"release_date"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type AlbumReq struct {
	Title       string    `json:"title" validate:"required"`
	Genre       string    `json:"genre" validate:"required"`
	ArtistID    string    `json:"artist_id" validate:"required"`
	ReleaseDate time.Time `json:"release_date"`
}

File: Python/django-songs-api/api/serializers/album.py
from rest_framework import serializers
from api.models import Album
from api.serializers.artist import ArtistSerializer


class AlbumSerializer(serializers.ModelSerializer):
    artist = ArtistSerializer(read_only=True)

    class Meta:
        model = Album
        fields = [
            'id',
            'title',
            'release_date',
            'artist',
            'created_at',
            'updated_at',
        ]
        read_only_fields = ['created_at', 'updated_at']